### PR TITLE
More setuptools.command.easy_install definitions.

### PR DESCRIPTION
**Pull request by ppentchev**
Hi,

Thanks a lot for maintaining and extending the type stubs in typeshed!

What do you think about this change that goes some of the way towards annotating the return types of some setuptools classes and methods? I am woefully aware that it is not complete, but it does help a project of mine that uses setuptools to parse the definitions of console script entry points.

Thanks in advance for your time, and keep up the great work!

G'luck,
Peter

### CODE REVIEW

File: stubs/setuptools/setuptools/command/easy_install.pyi [hunk 1]
```diff
@@ -110,15 +110,15 @@ class CommandSpec(list[str]):
     options: Any
     split_args: Any
     @classmethod
-    def best(cls): ...
+    def best(cls) -> CommandSpec: ...
```
Comment (Akuli) [inline stubs/setuptools/setuptools/command/easy_install.pyi:113]: This returns a class, not an instance.
```suggestion
    def best(cls) -> type[CommandSpec]: ...
```

File: stubs/setuptools/setuptools/command/easy_install.pyi [hunk 2]
```diff
@@ -127,17 +127,17 @@ class ScriptWriter:
     template: Any
     command_spec_class: Any
     @classmethod
-    def get_script_args(cls, dist, executable: Any | None = ..., wininst: bool = ...): ...
+    def get_script_args(cls, dist, executable: Any | None = ..., wininst: bool = ...) -> Iterable[tuple[str, str]]: ...
```
Comment (Akuli) [inline stubs/setuptools/setuptools/command/easy_install.pyi:130]: `Iterable` is correct, but `Iterator` or `Generator` might better, as that is more specific. Also applies to `get_args`.

File: stubs/setuptools/setuptools/command/easy_install.pyi [hunk 3]
```diff
@@ -127,17 +127,17 @@ class ScriptWriter:
...
+    def get_script_header(cls, script_text, executable: Any | None = ..., wininst: bool = ...) -> str: ...
     @classmethod
-    def get_args(cls, dist, header: Any | None = ...) -> None: ...
+    def get_args(cls, dist, header: Any | None = ...) -> Iterable[tuple[str, str]]: ...
     @classmethod
-    def get_writer(cls, force_windows): ...
+    def get_writer(cls, force_windows: bool) -> ScriptWriter: ...
     @classmethod
-    def best(cls): ...
+    def best(cls) -> ScriptWriter: ...
```
Comment (Akuli) [inline stubs/setuptools/setuptools/command/easy_install.pyi:138]:
```suggestion
    def best(cls) -> type[ScriptWriter]: ...
```

File: stubs/setuptools/setuptools/command/easy_install.pyi [hunk 4]
```diff
@@ -127,17 +127,17 @@ class ScriptWriter:
...
+    def get_script_args(cls, dist, executable: Any | None = ..., wininst: bool = ...) -> Iterable[tuple[str, str]]: ...
     @classmethod
-    def get_script_header(cls, script_text, executable: Any | None = ..., wininst: bool = ...): ...
+    def get_script_header(cls, script_text, executable: Any | None = ..., wininst: bool = ...) -> str: ...
     @classmethod
-    def get_args(cls, dist, header: Any | None = ...) -> None: ...
+    def get_args(cls, dist, header: Any | None = ...) -> Iterable[tuple[str, str]]: ...
     @classmethod
-    def get_writer(cls, force_windows): ...
+    def get_writer(cls, force_windows: bool) -> ScriptWriter: ...
```
Comment (Akuli) [inline stubs/setuptools/setuptools/command/easy_install.pyi:136]:
```suggestion
    def get_writer(cls, force_windows: bool) -> type[ScriptWriter]: ...
```

**Review by Akuli — COMMENTED** [review]
We appreciate the improvements :)

**Comment by srittau** [comment]
Ping @ppentchev

**Review by Akuli — APPROVED** [review]

**Pull request merged**